### PR TITLE
docs: Remove Google Group icon

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

In today's TSC meeting we agreed to retire the mailing list officially, so I removed the icon.

#### Is there anything you'd like reviewers to focus on?

Do the CSS changes make sense?

### CODE REVIEW

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving this open in case someone else wants to double-check the CSS changes.

**Review by amareshsm — COMMENTED** [review]
| Before | After |
|-------|-------|
| image | image |

Icons are not center aligned

File: docs/src/assets/scss/components/social-icons.scss [hunk 1]
```diff
@@ -7,10 +7,10 @@
         padding: 0;
         margin-left: -1rem;
         margin-inline-start: -1rem;
+        display: flex;
```
Comment (amareshsm) [inline docs/src/assets/scss/components/social-icons.scss:10]:
```suggestion
        display: flex;
        justify-content: center;
```

Either we can `justify-content: center` or we can completely remove the style changes and leave it as it is before.

**Comment by nzakas** [comment]
Thanks @amareshsm!

**Review by snitin315 — APPROVED** [review]
LGTM!

File: docs/src/assets/scss/components/social-icons.scss [hunk 2]
```diff
@@ -7,10 +7,11 @@
...
+        justify-content: center;
```
Comment (amareshsm) [inline docs/src/assets/scss/components/social-icons.scss:11]:
```suggestion
        display: inline-flex;
```
If we use `display: flex;` with `justify-content: center;` the items are not aligned towards the left in the desktop view.

  We can use `display: inline-flex;` it is working on screen sizes.

Comment (nzakas) [reply docs/src/assets/scss/components/social-icons.scss:11]: Hmmm, they look aligned to the left to me.
Screenshot 2023-01-20 at 10-39-52 Documentation - ESLint - Pluggable JavaScript Linter

Comment (amareshsm) [reply docs/src/assets/scss/components/social-icons.scss:11]: | Preview | Current (eslint.org) |
|---------|---------|
| Screenshot 2023-01-21 at 12 33 40 AM | Screenshot 2023-01-21 at 12 32 56 AM

**Pull request merged**